### ISSUE TITLE
Return status instead of throwing exception for expired license

### ISSUE THREAD

**Issue by xmenxwk** [LSXPrime/Aegis#13]
When we load license through LicenseManager.LoadLicenseAsync and if license is expired it will throw LicenseValidationException. Which seems all okay but to get new license we need to send request with expired license id. We can't have such request because license is not loaded and exception was thrown. So for validation fails or success, it should load license and returns a status enum. For file format and checksum and other file related exceptions should stay same.

One more thing
"SerialNumber" is being used, while its actually "LicenseKey", it would be better if one of them is choosen to avoid confusion.

**Comment by LSXPrime** [comment]
You're correct, throwing an exception for an expired license prevents accessing the `LicenseId` for renewal. I've updated `LicenseManager.LoadLicenseAsync` to return a `LicenseLoadResult` object instead. This object now includes a `LicenseStatus` enum (Valid, Expired, Revoked, Invalid) and the loaded `BaseLicense` even if validation fails, allowing you to access the `LicenseId` for renewal requests.

I will close the issue for now since this got addressed.
